src/generic/numberparsing.h

```diff
@@ -521,21 +1431,58 @@ really_inline bool parse_number(UNUSED const uint8_t *const src,
         // Ok, chances are good that we had an overflow!
         // this is almost never going to get called!!!
         // we start anew, going slowly!!!
-        return parse_float(src, parser, found_minus);
+        double d;
+        if(parse_float_strtod((const char*)src, &d)) {
```

Comment:
It is somewhat annoying here that we have to repeat the same parse_float_strtod code path twice. It would be nicer to find a way to have it once, thus reducing the number of lines of code.

Reply:
Done.